src/inmanta/resources.py

```diff
@@ -302,10 +414,15 @@ def map_field(
                 value = cls.map[field_name](exporter, model_object)
             else:
                 value = getattr(model_object, field_name)
-            # serialize to weed out all unknowns
-            # not very efficient, but the tree has to be traversed anyways
-            # passing along the serialized version would break the resource apis
-            json.dumps(value, default=inmanta.util.api_boundary_json_encoder)
+
+            # walk the entire model to find any value references and Unknowns (instead of json dump)
+            if reference_collector:
+                value = collect_references(reference_collector, value, field_name)
```

Comment:
does this cover all cases of the json collector?

Reply:
At least according to the comment. We can add more or even do an explicit serialization using our custom encoder